### ISSUE TITLE
Lycoris training ends up with blank lora

### ISSUE THREAD

**Issue by NaotoEnjoyer** [derrian-distro/LoRA_Easy_Training_Scripts#325]
hey i cant seem to train lycoris at all with this trainer, and jelosus' colab trainer, i use dev branch on both rn, issue is, when i try lycoris training, it looks like it doesnt update weights at all (scaled weight norm's avr key norm doesnt rise at all during training, while with basic lora and locon it does and loras work fine) 
here are the settings i tried to use, this is og preset where i didnt change paths and etc required stuff yet.. whats wrong with my training pls can you help me
[settings.txt](https://github.com/user-attachments/files/20645227/settings.txt)

**Comment by derrian-distro** [comment]
The settings dont seem like they would completely break anything, and with my base settings with some tweaks for testing, lycoris locon works fine for me, unfortunately I cannot help any more right this moment as I am without a computer for a bit

**Comment by NaotoEnjoyer** [comment]
tyvm for giving the attention none the less, can you pls explain me what might be the cause? im using jelosus' colab trainer with -b dev on your repo clone line and dev on training scripts local app, maybe i can upload entire training log or it wouldnt be helpful? idk ppl who looked at it cant say whats wrong everything seems to be fine, tho ais like gemini said that might be because of algo not being stated properly, which freezes lycoris up in training

**Comment by NaotoEnjoyer** [comment]
can you pls upload me your config and say what setup exactly did you test lycoris with? like one i described

**Comment by derrian-distro** [comment]
Unfortunately I cannot provide a config, I am *still* without a computer and will likely continue to be for at least another day or two

**Comment by NaotoEnjoyer** [comment]
here is how training starts on the colab btw, maybe there might be some clue?:
/content/trainer/sd_scripts/venv/lib/python3.10/site-packages/diffusers/utils/outputs.py:63: FutureWarning: `torch.utils._pytree._register_pytree_node` is deprecated. Please use `torch.utils._pytree.register_pytree_node` instead.
  torch.utils._pytree._register_pytree_node(
/content/trainer/sd_scripts/venv/lib/python3.10/site-packages/diffusers/utils/outputs.py:63: FutureWarning: `torch.utils._pytree._register_pytree_node` is deprecated. Please use `torch.utils._pytree.register_pytree_node` instead.
  torch.utils._pytree._register_pytree_node(
rich is not installed, using basic logging
Loading settings from /content/trainer/runtime_store/config.toml...
tokenizer_config.json: 100% 905/905 [00:00<00:00, 4.72MB/s]
vocab.json: 100% 961k/961k [00:00<00:00, 1.44MB/s]
merges.txt: 100% 525k/525k [00:00<00:00, 2.35MB/s]
special_tokens_map.json: 100% 389/389 [00:00<00:00, 2.36MB/s]
tokenizer.json: 100% 2.22M/2.22M [00:01<00:00, 2.06MB/s]
/content/trainer/sd_scripts/venv/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884
  warnings.warn(
tokenizer_config.json: 100% 904/904 [00:00<00:00, 3.25MB/s]
vocab.json: 100% 862k/862k [00:00<00:00, 93.7MB/s]
merges.txt: 100% 525k/525k [00:00<00:00, 15.2MB/s]
special_tokens_map.json: 100% 389/389 [00:00<00:00, 2.07MB/s]
tokenizer.json: 100% 2.22M/2.22M [00:01<00:00, 2.00MB/s]
Loading dataset config from /content/trainer/runtime_store/dataset.toml
prepare images.
get image size from name of cache files
100% 105/105 [00:00<00:00, 554662.37it/s]
set image size from cache files: 0/105
found directory /content/drive/MyDrive/Lora/dataset contains 105 image files
read caption: 100% 105/105 [00:00<00:00, 520.71it/s]
420 train images with repeats.
0 reg images with repeats.
no regularization images / 正則化画像が見つかりませんでした
[Dataset 0]
  batch_size: 6
  resolution: (1024, 1024)
  resize_interpolation: None
  enable_bucket: True
  min_bucket_reso: 256
  max_bucket_reso: 8192
  bucket_reso_steps: 64
  bucket_no_upscale: False

  [Subset 0 of Dataset 0]
    image_dir: "/content/drive/MyDrive/Lora/dataset"
    image_count: 105
    num_repeats: 4
    shuffle_caption: True
    keep_tokens: 0
    caption_dropout_rate: 0.0
    caption_dropout_every_n_epochs: 0
    caption_tag_dropout_rate: 0.0
    caption_prefix: None
    caption_suffix: None
    color_aug: False
    flip_aug: False
    face_crop_aug_range: None
    random_crop: False
    token_warmup_min: 1,
    token_warmup_step: 0,
    alpha_mask: False
    resize_interpolation: None
    custom_attributes: {}
    is_reg: False
    class_tokens: None
    caption_extension: .txt


[Prepare dataset 0]
loading image sizes.
100% 105/105 [01:34<00:00,  1.12it/s]
make buckets
number of images (including repeats) / 各bucketの画像枚数（繰り返し回数を含む）
bucket 0: resolution (832, 1216), count: 404
bucket 1: resolution (960, 1088), count: 4
bucket 2: resolution (1152, 896), count: 4
bucket 3: resolution (1216, 832), count: 8
mean ar error (without repeats): 0.02984842350115133
preparing accelerator
/content/trainer/sd_scripts/venv/lib/python3.10/site-packages/accelerate/accelerator.py:488: FutureWarning: `torch.cuda.amp.GradScaler(args...)` is deprecated. Please use `torch.amp.GradScaler('cuda', args...)` instead.
  self.scaler = torch.cuda.amp.GradScaler(**kwargs)
accelerator device: cuda
loading model for process 0/1
load StableDiffusion checkpoint: /content/pretrained_model/Illustrious-XL-v0.1.safetensors
building U-Net
loading U-Net from checkpoint
U-Net: <All keys matched successfully>
building text encoders
loading text encoders from checkpoint
text encoder 1: <All keys matched successfully>
text encoder 2: <All keys matched successfully>
building VAE
loading VAE from checkpoint
VAE: <All keys matched successfully>
load VAE: /content/vae/sdxl_vae.safetensors
additional VAE loaded
Enable SDPA for U-Net
import network module: lycoris.kohya
[Dataset 0]
caching latents with caching strategy.
caching latents...
100% 105/105 [03:32<00:00,  2.02s/it]
2025-06-10 09:53:28|[LyCORIS]-INFO: Using rank adaptation algo: locon
2025-06-10 09:53:28|[LyCORIS]-INFO: Use Dropout value: 0.0
2025-06-10 09:53:28|[LyCORIS]-INFO: Use LORA Dropout value: 0.0
2025-06-10 09:53:28|[LyCORIS]-INFO: Use AID Dropout value: 0.0
2025-06-10 09:53:28|[LyCORIS]-INFO: wd_on_output=False
2025-06-10 09:53:28|[LyCORIS]-INFO: Create LyCORIS Module
2025-06-10 09:53:28|[LyCORIS]-INFO: Create LyCORIS Module
2025-06-10 09:53:28|[LyCORIS]-INFO: create LyCORIS for Text Encoder: 264 modules.
2025-06-10 09:53:28|[LyCORIS]-INFO: Create LyCORIS Module
2025-06-10 09:53:30|[LyCORIS]-INFO: create LyCORIS for U-Net: 788 modules.
2025-06-10 09:53:30|[LyCORIS]-INFO: module type table: {'LoConModule': 1052}
2025-06-10 09:53:30|[LyCORIS]-INFO: enable LyCORIS for U-Net
prepare optimizer, data loader etc.
use came_pytorch.CAME.CAME | {'weight_decay': 0.1, 'betas': (0.9, 0.999, 0.99995)}
use LoraEasyCustomOptimizer.RexAnnealingWarmRestarts.RexAnnealingWarmRestarts | {'min_lr': 1e-07, 'gamma': 0.9, 'd': 0.9, 'warmup_steps': 52, 'first_cycle_max_steps': 350} as lr_scheduler
enable full fp16 training.
running training / 学習開始
  num train images * repeats / 学習画像の数×繰り返し回数: 420
  num validation images * repeats / 学習画像の数×繰り返し回数: 0
  num reg images / 正則化画像の数: 0
  num batches per epoch / 1epochのバッチ数: 72
  num epochs / epoch数: 20
  batch size per device / バッチサイズ: 6
  gradient accumulation steps / 勾配を合計するステップ数 = 4
  total optimization steps / 学習ステップ数: 350
unet dtype: torch.float16, device: cuda:0
text_encoder [0] dtype: torch.float16, device: cuda:0
text_encoder [1] dtype: torch.float16, device: cuda:0
steps:   0% 0/350 [00:00<?, ?it/s]
epoch 1/20

epoch is incremented. current_epoch: 0, epoch: 1
steps:   0% 1/350 [00:52<5:02:41, 52.04s/it, Average key norm=tensor(7.3016e-05, device='cuda:0', dtype=torch.float16), Keys Scaled=0, avr_loss=0.124]

**Comment by NaotoEnjoyer** [comment]
yeah it still produces whole bunch of nothing.. i tried both dev branch and main branch of trainer repo, and app repo, also can you pls apply adamw8bit bitsandbytes fix to main branch too?

**Comment by Jelosus2** [comment]
Closing this as it was solved
